addons/account_payment/views/account_portal_templates.xml

```diff
@@ -71,7 +71,7 @@
                     <h4 class="mb24">Pay with</h4>
                     <t t-call="payment.payment_tokens_list">
                         <t t-set="mode" t-value="'payment'"/>
-                        <t t-set="partner_id" t-value="request.env.user.partner_id.id"/>
+                        <t t-set="partner_id" t-value="invoice.partner_id.id if request.env.user == request.env.ref('base.public_user') else request.env.user.partner_id.id"/>
```

Comment:
I think it is better to call `_is_public` on the current user as opposed to comparing users?

Reply:
Yeah right, I didnot realize it existed in 11.0 :D